Sorter already used

```
[Chunk Render Task Executor #0/ERROR]: Chunk build failed
java.lang.IllegalStateException: Sorter already used!
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.translucent_sorting.data.StaticNormalRelativeData.getSorter(StaticNormalRelativeData.java:32) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.compile.tasks.ChunkBuilderMeshingTask.execute(ChunkBuilderMeshingTask.java:219) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.compile.tasks.ChunkBuilderMeshingTask.execute(ChunkBuilderMeshingTask.java:52) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.compile.executor.ChunkJobTyped.execute(ChunkJobTyped.java:47) [sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.compile.executor.ChunkBuilder$WorkerRunnable.run(ChunkBuilder.java:193) [sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1447) [?:?]
[Render thread/ERROR]: Unreported exception thrown!
java.lang.RuntimeException: Exception thrown while executing job
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.compile.executor.ChunkJobResult.unwrap(ChunkJobResult.java:34) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.RenderSectionManager.collectChunkBuildResults(RenderSectionManager.java:471) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.RenderSectionManager.uploadChunks(RenderSectionManager.java:355) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.SodiumWorldRenderer.setupTerrain(SodiumWorldRenderer.java:243) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot//MC/net.minecraft.client.render.WorldRenderer.setupTerrain(WorldRenderer.java:2670) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.render.WorldRenderer.render(WorldRenderer.java:470) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.render.GameRenderer.renderWorld(GameRenderer.java:738) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.render.GameRenderer.render(GameRenderer.java:506) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.MinecraftClient.render(MinecraftClient.java:1361) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.MinecraftClient.run(MinecraftClient.java:947) [client-intermediary.jar:?]
	at knot/net.minecraft.client.main.Main.main(Main.java:265) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:506) [fabric-loader-0.17.2.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:72) [fabric-loader-0.17.2.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.17.2.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:105) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?]
Caused by: java.lang.IllegalStateException: Sorter already used!
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.translucent_sorting.data.StaticNormalRelativeData.getSorter(StaticNormalRelativeData.java:32) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.compile.tasks.ChunkBuilderMeshingTask.execute(ChunkBuilderMeshingTask.java:219) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.compile.tasks.ChunkBuilderMeshingTask.execute(ChunkBuilderMeshingTask.java:52) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.compile.executor.ChunkJobTyped.execute(ChunkJobTyped.java:47) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at knot/net.caffeinemc.mods.sodium.client.render.chunk.compile.executor.ChunkBuilder$WorkerRunnable.run(ChunkBuilder.java:193) ~[sodium-fabric-0.7.0-snapshot+mc1.21.8-build.707.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1447) ~[?:?]
```

--- a/common/src/main/java/net/caffeinemc/mods/sodium/client/render/chunk/compile/tasks/ChunkBuilderMeshingTask.java
+++ b/common/src/main/java/net/caffeinemc/mods/sodium/client/render/chunk/compile/tasks/ChunkBuilderMeshingTask.java
@@ -18,6 +18,7 @@ import net.caffeinemc.mods.sodium.client.render.chunk.terrain.material.DefaultMa
 import net.caffeinemc.mods.sodium.client.render.chunk.translucent_sorting.SortBehavior;
 import net.caffeinemc.mods.sodium.client.render.chunk.translucent_sorting.SortType;
 import net.caffeinemc.mods.sodium.client.render.chunk.translucent_sorting.TranslucentGeometryCollector;
+import net.caffeinemc.mods.sodium.client.render.chunk.translucent_sorting.data.DynamicData;
 import net.caffeinemc.mods.sodium.client.render.chunk.translucent_sorting.data.PresentTranslucentData;
 import net.caffeinemc.mods.sodium.client.render.chunk.translucent_sorting.data.TranslucentData;
 import net.caffeinemc.mods.sodium.client.services.PlatformLevelRenderHooks;
@@ -174,7 +175,15 @@ public class ChunkBuilderMeshingTask extends ChunkBuilderTask<ChunkBuildOutput> 
         boolean reuseUploadedData = false;
         TranslucentData translucentData = null;
         if (sortEnabled) {
-            var oldData = this.render.getTranslucentData();
+            TranslucentData oldData = this.render.getTranslucentData();
+            
+            // Reusing non-dynamic data leads to attempting to sort with it again,
+            // which throws an exception since it can only generate a sorter once.
+            // To prevent this, reusing data is prevented when forceSort is enabled and the data is not dynamic.
+            if (this.forceSort && !(oldData instanceof DynamicData)) {
+                oldData = null;
+            }
+            
             translucentData = collector.getTranslucentData(oldData, this);
             reuseUploadedData = !this.forceSort && translucentData == oldData;
         }

